src/period/helpers.js

```diff
@@ -149,3 +149,35 @@ export function getFirstDateOfWeek(year, week) {
 
     return new Date(year, month, ordDate - ordDiff[month]);
 }
+
+export const computeWeekBasedPeriod = ({ year, week, locale = 'en', weekTypeDiff = 0, periodLength = 6 }) => {
```

Comment:
This function is very confusing to me. Even after reading the code, it's very, very difficult to understand what's happening here. Can you add a doc block with some examples, especially for the 53rd week edge case?

Reply:
Lemme know if my explanation helps

Reply:
Sorry for not making clear what I meant.. my bad!
I was talking about the `@example` jsdoc comment